### PR TITLE
[9459] show a useful traceback when awaiting a traceback-bearing failed Deferred

### PR DESCRIPTION
**Pull request by glyph**
## Contributor Checklist:

* [x] There is an associated ticket in Trac. https://twistedmatrix.com/trac/ticket/9459
* [x] The changes pass minimal style checks (see: https://twistedmatrix.com/trac/wiki/TwistedDevelopment#GettingYourPatchAccepted )
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles )
* [x] I have updated the automated tests.

### CODE REVIEW

**Comment by hawkowl** [comment]
@glyph Twistedchecker is angry:
```
src/twisted/internet/test/test_coroutines.py (0.0%):
    18: C0301: (line-too-long), : Line too long (80/79)
    41: C0301: (line-too-long), : Line too long (84/79)
```

**Comment by glyph** [comment]
Thanks for the heads up @hawkowl , let's see if I can get this fixed before lunch :)

**Comment by glyph** [comment]
/me watches the wall clock

have we considered switching the twistedchecker environment to use pypy

**Comment by glyph** [comment]
@hawkowl GREEN CHECK SHIP IT

**Comment by hawkowl** [comment]
but... i haven't even read the code yet D:

tomorrow ™️

**Comment by glyph** [comment]
🦉 🕚

File: src/twisted/newsfragments/9459.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Failure tracebacks generated by coroutines scheduled with twisted.internet.defer.ensureDeferred - i.e. any Deferred-awaiting coroutine - now contain fewer extraneous frames from the trampoline implementation, and correctly indicate the source of exceptions raised in other call stacks - i.e. the function that raised the exception.  In other words: if you 'await' a function that raises an exception, you'll be able to see where the error came from.
```
Comment (hawkowl) [inline src/twisted/newsfragments/9459.feature]: itym twisted.python.failure.Failure :)

**Comment by hawkowl** [comment]
not including inlinecallbacks garbage in tracebacks :ok_hand: :ok_hand: :ok_hand: :ok_hand: :ok_hand: :ok_hand: :ok_hand: :ok_hand: :ok_hand: :ok_hand: :ok_hand: :100: :100: :100: :100: :100: :100:

**Review by hawkowl — APPROVED** [review]
newsfile nit which maybe makes it less sensical but is to the standard which we might want to ignore, otherwise, it appears to all work as intended.

**Comment by glyph** [comment]
I think I am going to be optimistic and assume that the tests will pass since the only change from a passing build was the newsfile content.

**Pull request merged**